google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -366,9 +366,15 @@ public GapicSpannerRpc(final SpannerOptions options) {
       // If it is enabled in options uses the channel pool provided by the gRPC-GCP extension.
       maybeEnableGrpcGcpExtension(defaultChannelProviderBuilder, options);
 
+      InstantiatingGrpcChannelProvider defaultChannelProvider =
+          defaultChannelProviderBuilder.build();
+
       TransportChannelProvider channelProvider =
-          MoreObjects.firstNonNull(
-              options.getChannelProvider(), defaultChannelProviderBuilder.build());
+          MoreObjects.firstNonNull(options.getChannelProvider(), defaultChannelProvider);
+
+      boolean isDirectPathChannelCreated =
+          defaultChannelProvider.canUseDirectPath()
+              && defaultChannelProvider.isDirectPathXdsEnabled();
```

Comment:
I think that this should be:

```suggestion
      boolean isDirectPathChannelCreated =
          options.getChannelProvider() == null && defaultChannelProvider.canUseDirectPath()
              && defaultChannelProvider.isDirectPathXdsEnabled();
```